k8s/upgrade/src/plugin/upgrade.rs

```diff
@@ -238,31 +317,43 @@ impl UpgradeEventClient {
             .collect::<Vec<_>>();
 
         event_list.sort_by(|a, b| b.event_time.cmp(&a.event_time));
-        match event_list.get(0) {
-            Some(event) => match event.message.clone() {
-                Some(data) => {
-                    let e: UpgradeEvent = serde_json::from_str(data.as_str())
-                        .context(error::EventSerdeDeserialization { event: data })?;
-                    println!("Upgrade From: {}", e.from_version);
-                    println!("Upgrade To: {}", e.to_version);
-                    println!("Upgrade Status: {}", e.message);
-                    Ok(())
-                }
-                None => error::MessageInEventNotPresent.fail(),
-            },
-            None => error::UpgradeEventNotPresent.fail(),
+        let latest_event = event_list
+            .get(0)
```

Comment:
hmm are we returning the latest event of the first?

Reply:
There can be multiple events so sorting them and getting the first element gives the latest event .

Reply:
Ah I would have thought sorting by timestamp would give 0 as the earliest rather than latest.
Btw when we get a validation error there's no chance of another event being added after that?

Reply:
No , once we get a validation we error out and return. So no other upgrade related event can be publihsed
```
   if let Err(error) = helm_upgrade_dry_run.dry_run().await {
        event.publish_unrecoverable(&error, true).await;
        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
        return Err(error);
    }
```

Reply:
So the job doesn't get restarted again on failure?